packages/dev/core/src/Misc/greasedLineTools.ts

```diff
@@ -524,8 +524,14 @@ export class GreasedLineTools {
      * @returns the colors texture
      */
     public static CreateColorsTexture(name: string, colors: Color3[], colorsSampling: number, scene: Scene) {
+        const maxTextureSize = scene?.getEngine()?.getCaps().maxTextureSize ?? 1;

```

Comment:
`scene` can't be null/empty, you should write it like `scene.getEngine().getCaps().maxTextureSize`.